treeherder/log_parser/utils.py

```diff
@@ -1,66 +1,31 @@
-import logging
-
 import simplejson as json
-from requests.exceptions import HTTPError
-
-from treeherder.etl.artifact import (serialize_artifact_json_blobs,
-                                     store_job_artifacts)
-from treeherder.log_parser.artifactbuildercollection import (ArtifactBuilderCollection,
-                                                             LogSizeException)
-from treeherder.model.models import JobLog
-
-logger = logging.getLogger(__name__)
-
-
-def extract_text_log_artifacts(job_log):
-    """Generate a set of artifacts by parsing from the raw text log."""
-
-    # parse a log given its url
-    artifact_bc = ArtifactBuilderCollection(job_log.url)
-    artifact_bc.parse()
-
-    artifact_list = []
-    for name, artifact in artifact_bc.artifacts.items():
-        artifact_list.append({
-            "job_guid": job_log.job.guid,
-            "name": name,
-            "type": 'json',
-            "blob": json.dumps(artifact)
-        })
+from jsonschema import (ValidationError,
+                        validate)
 
-    return artifact_list
+with open('schemas/performance-artifact.json') as f:
+    PERFHERDER_SCHEMA = json.load(f)
 
 
-def post_log_artifacts(job_log):
-    """Post a list of artifacts to a job."""
-    logger.debug("Downloading/parsing log for log %s", job_log.id)
+def validate_perf_data(performance_data: dict):
+    validate(performance_data, PERFHERDER_SCHEMA)
+    expected_range = (45, PERFHERDER_SCHEMA["definitions"]["suite_schema"]["properties"]["extraOptions"]["items"]["maxLength"])
+    # allow only one extraOption longer than 45
+    if len(_long_elements(expected_range, _extra_options(performance_data))) > 1:
+        raise ValidationError("Too long extraOption. Max: {}".format(expected_range[0]))
 
-    try:
-        artifact_list = extract_text_log_artifacts(job_log)
-    except LogSizeException as e:
-        job_log.update_status(JobLog.SKIPPED_SIZE)
-        logger.warning('Skipping parsing log for %s: %s', job_log.id, e)
-        return
-    except Exception as e:
-        job_log.update_status(JobLog.FAILED)
 
-        # Unrecoverable http error (doesn't exist or permission denied).
-        # Apparently this can happen somewhat often with taskcluster if
-        # the job fails (bug 1154248), so just warn rather than raising,
-        # to prevent the noise/load from retrying.
-        if isinstance(e, HTTPError) and e.response.status_code in (403, 404):
-            logger.warning("Unable to retrieve log for %s: %s", job_log.id, e)
-            return
+def _long_elements(expected_range: tuple, all_extra_options: list):
+    long_elements = []
+    for list_item in all_extra_options:
+        for element in list_item:
+            if len(element) in range(expected_range[0] + 1, expected_range[1] + 1):
```

Comment:
This is hard on the eyes.
Something like:
```python
 if expected_range[0] + 1 < len(element) < expected_range[1] + 1:
```
is a tad better. Though we still need to improve the looking of the `expected_range...+1`  expressions

Reply:
If you agree, I will replace expected_range with two parameters, min and max.
The new signature:    def _long_elements(all_extra_options: list, min: int, max: int):
The calling:   if len(_long_elements(_extra_options(suite), *expected_range)) > 1:
The check:   if len(element) in range(min + 1, max + 1):

Do you agree with this aproach?

Reply:
Yes, it's better this way.